docs/userguide/mng-storage-driver.md

```diff
@@ -0,0 +1,145 @@
+container has many changes to an image, AUFS has to build and then traverse many
+branches in the containers file system.
+
+If you use AUFS as your storage backend, it is good practice to put important data in volumes and mount them into your container.
+
+## Device Mapper
+
+The Device Mapper storage driver was contributed by RedHat. It is a complex subsystem that supports RAID, disk encryption, and snapshots. This is the default storage driver on RedHat/CentOS distributions.
+
+The Device Mapper manages changes on the **block** level using thin
+provisioning. A thin pool is created based on two block devices, one for data
+and one for metadata. The data is called the "pool" and is just big pool of blocks. Docker uses the smallest possible block size of 64 KB.
+
+Each change allocates a new block from the pool.  When there are no more blocks
+in the pool, writes stall until more blocks or allocated or the write times out.
+At any give time, it is possible to take a snapshot of an existing container to
+image. You can use the container snapshot to create a frozen image.  You can
+create a container from an image snapshot.
+
+Since the Device Mapper operates on the block level, diffs between images and
+containers are not as visible as other storage drivers.  Performance is
+reasonable but not as fast as AUFS. Best performance is achieved by using the
+storage driver options to put data and metadata on real devices. You should also
+familiarize yourself with the options for tuning the pool.
+
+## BTRFS
+The BTFS driver integrates the snapshot and block pool management features at the file system level instead of the block device level. If the Docker host does not include AUFS and BTFS is available, then it is the default storage driver.
+
+BTFS stores changes in chunks that contain data or metadata. While there is no pool to empty, the chunks can fill up the available space. The result is a  `No space level on device` error which requires you to intervene. BTFS is fast and stable like AUFS. It doesn't require a lot of tuning.
+
+## ZFS
+The `zfs` driver is probably not fast as `btrfs` but has a longer track record
+on stability. Thanks to `Single Copy ARC` shared blocks between clones will be
+cached only once. Use `docker daemon -s zfs`. To select a different zfs filesystem
+set `zfs.fsname` option as described in [Storage driver options](#storage-driver-options).
+
+
+## Overlayfs
+
+The `overlay` is a very fast union filesystem. It is now merged in the main
+Linux kernel as of [3.18.0](https://lkml.org/lkml/2014/10/26/137). Call
+`docker daemon -s overlay` to use it.
+
+> **Note:**
+> As promising as `overlay` is, the feature is still quite young and should not
+> be used in production. Most notably, using `overlay` can cause excessive
+> inode consumption (especially as the number of images grows), as well as
+> being incompatible with the use of RPMs.
+
+> **Note:**
+> It is currently unsupported on `btrfs` or any Copy on Write filesystem
```

Comment:
Also *none* of the CoW filesystems is supported on another CoW backing filesystem (afaik)